src/main/java/uk/gov/pay/webhooks/message/WebhookMessageService.java

```diff
@@ -64,17 +65,24 @@ public void handleInternalEvent(InternalEvent event) throws IOException, Interru
             subscribedWebhooks
                     .stream()
                     .map(webhook -> buildWebhookMessage(webhook, event, ledgerTransaction))
+                    .flatMap(Optional::stream)
                     .map(webhookMessageDao::create)
                     .forEach(webhookMessageEntity -> webhookDeliveryQueueDao.enqueueFrom(webhookMessageEntity, WebhookDeliveryQueueEntity.DeliveryStatus.PENDING, Date.from(instantSource.instant())));
         }
     }
 
-    private WebhookMessageEntity buildWebhookMessage(WebhookEntity webhook, InternalEvent event, LedgerTransaction ledgerTransaction) {
+    private Optional<WebhookMessageEntity> buildWebhookMessage(WebhookEntity webhook, InternalEvent event, LedgerTransaction ledgerTransaction) {
         JsonNode resource = switch (event.resourceType()) {
             case "payment" -> objectMapper.valueToTree(PaymentApiRepresentation.of(ledgerTransaction));
             case "refund" -> objectMapper.valueToTree(RefundApiRepresentation.of(ledgerTransaction));
-            default -> objectMapper.valueToTree(ledgerTransaction);
+            default -> null;
```

Comment:
Hmm, feels a bit like we’re using null as a magic value. Since we actually want to change the flow control, maybe we should just use a switch statement, which could be done like this:

```java
private Optional<WebhookMessageEntity> buildWebhookMessage(WebhookEntity webhook, InternalEvent event, LedgerTransaction ledgerTransaction) {
    switch (event.resourceType()) {
        case "payment" -> {
            JsonNode resource = objectMapper.valueToTree(PaymentApiRepresentation.of(ledgerTransaction));
            return Optional.of(buildWebhookMessageEntity(webhook, event, resource));
        }
        case "refund" -> {
            JsonNode resource = objectMapper.valueToTree(RefundApiRepresentation.of(ledgerTransaction));
            return Optional.of(buildWebhookMessageEntity(webhook, event, resource));
        }
        default -> {
            LOGGER.info("Ignoring unsupported resource type %s".formatted(event.resourceType()));
            return Optional.empty();
        }
    }
}

private WebhookMessageEntity buildWebhookMessageEntity(WebhookEntity webhook, InternalEvent event, JsonNode resource) {
    var webhookMessageEntity = new WebhookMessageEntity();
    // …
    return webhookMessageEntity;
}
```